subsys/bluetooth/audio/stream.c

```diff
@@ -975,6 +984,84 @@ static void unicast_group_free(struct bt_audio_unicast_group *group)
 	group->allocated = false;
 }
 
+static int stream_param_check(struct bt_audio_unicast_group_stream_param *param)
+{
+	CHECKIF(param->stream == NULL) {
+		LOG_ERR("param->stream is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(param->qos == NULL) {
+		LOG_ERR("param->qos is NULL");
+		return -EINVAL;
+	}
+
+	if (param->stream != NULL && param->stream->group != NULL) {
+		LOG_WRN("stream %p already part of group %p",
+			param->stream, param->stream->group);
+		return -EALREADY;
+	}
+
+	CHECKIF(!bt_audio_valid_qos(param->qos)) {
+		LOG_ERR("Invalid QoS");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stream_pair_param_check(struct bt_audio_unicast_group_stream_pair_param *param)
+{
+	int err;
+
+	CHECKIF(param->rx_param == NULL && param->tx_param == NULL) {
+		LOG_DBG("Invalid stream parameters");
+		return -EINVAL;
+	}
+
+	if (param->rx_param != NULL) {
+		err = stream_param_check(param->rx_param);
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	if (param->tx_param != NULL) {
+		err = stream_param_check(param->tx_param);
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static int group_qos_common_set(const struct bt_codec_qos **group_qos,
+				struct bt_audio_unicast_group_stream_pair_param *param)
+{
+	if (param->rx_param != NULL) {
+		if (*group_qos == NULL) {
+			*group_qos = param->rx_param->qos;
+		} else if (!unicast_group_valid_qos(*group_qos, param->rx_param->qos)) {
```

Comment:
This function was removed in merged https://github.com/zephyrproject-rtos/zephyr/pull/53119